src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTime.cs

```diff
@@ -0,0 +1,278 @@
+        /// Thrown if this would result in an invalid JSON to be written (while validation is enabled).
+        /// </exception>
+        public void WriteString(ReadOnlySpan<byte> propertyName, DateTime value, bool suppressEscaping = false)
+        {
+            JsonWriterHelper.ValidateProperty(ref propertyName);
+
+            if (!suppressEscaping)
+                WriteStringSuppressFalse(ref propertyName, value);
+            else
+                WriteStringByOptions(ref propertyName, value);
+
+            _currentDepth |= 1 << 31;
+            _tokenType = JsonTokenType.String;
+        }
+
+        private void WriteStringSuppressFalse(ref ReadOnlySpan<char> propertyName, DateTime value)
+        {
+            int propertyIdx = JsonWriterHelper.NeedsEscaping(propertyName);
+
+            Debug.Assert(propertyIdx >= -1 && propertyIdx < int.MaxValue / 2);
+
+            if (propertyIdx != -1)
+            {
+                WriteStringEscapeProperty(ref propertyName, value, propertyIdx);
+            }
+            else
+            {
+                WriteStringByOptions(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringSuppressFalse(ref ReadOnlySpan<byte> propertyName, DateTime value)
+        {
+            int propertyIdx = JsonWriterHelper.NeedsEscaping(propertyName);
+
+            Debug.Assert(propertyIdx >= -1 && propertyIdx < int.MaxValue / 2);
+
+            if (propertyIdx != -1)
+            {
+                WriteStringEscapeProperty(ref propertyName, value, propertyIdx);
+            }
+            else
+            {
+                WriteStringByOptions(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringEscapeProperty(ref ReadOnlySpan<char> propertyName, DateTime value, int firstEscapeIndexProp)
+        {
+            Debug.Assert(int.MaxValue / MaxExpansionFactorWhileEscaping >= propertyName.Length);
+
+            char[] propertyArray = null;
+
+            int length = firstEscapeIndexProp + MaxExpansionFactorWhileEscaping * (propertyName.Length - firstEscapeIndexProp);
+            Span<char> span;
+            if (length > StackallocThreshold)
+            {
+                propertyArray = ArrayPool<char>.Shared.Rent(length);
+                span = propertyArray;
+            }
+            else
+            {
+                // Cannot create a span directly since the span gets exposed outside this method.
+                unsafe
+                {
+                    char* ptr = stackalloc char[length];
+                    span = new Span<char>(ptr, length);
+                }
+            }
+            JsonWriterHelper.EscapeString(ref propertyName, ref span, firstEscapeIndexProp, out int written);
+            propertyName = span.Slice(0, written);
+
+            WriteStringByOptions(ref propertyName, value);
+
+            if (propertyArray != null)
+                ArrayPool<char>.Shared.Return(propertyArray);
```

Comment:
Do we care if this (or similar methods) throw and the rented array is not returned?